### PR TITLE
        rgw:when call the multiparts upload API TransferManager.upload() by s3 java sdk , got the SignatureDoesNotMatch except

### PR DESCRIPTION
**Pull request by BodihTao**
fixbug: http://tracker.ceph.com/issues/18355

### CODE REVIEW

**Comment by hjwsm1989** [comment]
is the second commit necessary?

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -3730,18 +3730,11 @@ int RGW_Auth_S3::authorize_v4(RGWRados *store, struct req_state *s, bool force_b
         if (key != "X-Amz-Credential") {
           string key_decoded;
           url_decode(key, key_decoded);
-          if (key.length() != key_decoded.length()) {
-            encoded_key = key;
-          } else {
-            aws4_uri_encode(key, encoded_key);
-          }
+	  aws4_uri_encode(key_decoded, encoded_key);
```
Comment (amitkumar50) [inline src/rgw/rgw_rest_s3.cc]: @BodihTao (NIT)
https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits

Comment (amitkumar50) [inline src/rgw/rgw_rest_s3.cc]: @BodihTao Can you plz rebase it?

**Review by murolo — COMMENTED** [review]
how dose it works

**Pull request closed without merging**